Sources/Amplitude/Plugins/iOS/iOSUserInteractionCapture/CustomUIGestureRecognizers.swift

```diff
@@ -0,0 +1,312 @@
+            $0.type.contains(.adjustable)
+        }),
+           let slider = target.object as? UISlider
+        {
+            initialTarget = target
+            initialValue = slider.value
+        }
+    }
+
+    override func touchesEnded(_ touches: Set<UITouch>, with event: UIEvent) {
+        super.touchesEnded(touches, with: event)
+
+        guard
+            let captureDelegate = captureDelegateHandler()
+        else {
+            return
+        }
+
+        defer {
+            self.initialTarget = nil
+            self.initialValue = .zero
+        }
+
+        guard
+            let initialTarget,
+            let initialValue,
+            let slider = initialTarget.object as? UISlider,
+            slider.value != initialValue,
+            abs(initialValue - slider.value) >= GlobalUISlideGestureRecognizer.minimumSlideAmount
+        else {
+            return
+        }
+
+        let percentage = Int(round(slider.value * 100))
+
+        captureDelegate.amplitude?.track(event: UserInteractionEvent(
+            .sliderChanged(to: percentage),
+            label: initialTarget.label,
+            value: initialTarget.value,
+            type: initialTarget.type))
+    }
+}
+
+// MARK: -
+
+internal final class GlobalUIPressGestureRecognizer: UIGestureRecognizer {
+
+    // MARK: - Private Properties
+
+    /// The activation delta between the initial touch point and the final touch point.
+    private static let pressActivationDelta: Double = 10
+
+    /// The minimum press duration to detect a long press.
+    private static let minimumPressDuration: TimeInterval = 0.5
+
+    /// The number of clicks required to classify the followed clicks as rage clicks during
+    /// the `rageClickTimeWindow` period.
+    private static let rageClickCountThreshold = 5
+
+    /// The time window to detect rage clicks.
+    private static let rageClickTimeWindow: TimeInterval = 0.8
+
+    /// The delegate that manages user interactions options.
+    private var captureDelegateHandler: () -> UserInteractionCaptureDelegate?
+
+    /// The target element when the screen was first touched.
+    private var initialTarget: AccessibilityTarget?
+
+    /// The root view of the screen when it was first touched.
+    private var initialRootView: UIView?
+
+    /// The initial point where the scroll gesture began.
+    private var initialTouchLocation: CGPoint?
+
+    /// The start time of the press gesture.
+    private var startTime: TimeInterval?
+
+    /// Specifies whether the followed clicks are rage clicks.
+    private var isRageClicking = false
+
+    /// The element targetted by a series of rage clicks.
+    private var currentRageClickTarget: NSObject?
+
+    /// Recent taps and the corresponding time associated with the tap.
+    private var recentTaps: [(target: AccessibilityTarget, time: TimeInterval)] = []
+
+    // MARK: - Life Cycle
+
+    init(for captureDelegateHandler: @escaping () -> UserInteractionCaptureDelegate?) {
```

Comment:
I was trying to get a weak reference to self. A capture list within a closure seemed to be the only way to do this. Is there a better way to accomplish this? (the weak reference here was to prevent `CustomUIGestureRecognizers` and `UserInteractionCaptureDelegate` have circular reference.)

Reply:
Ah, this is probably a good time to learn the delegate pattern :) - see https://developer.apple.com/library/archive/documentation/General/Conceptual/CocoaEncyclopedia/DelegatesandDataSources/DelegatesandDataSources.html, or maybe https://medium.com/@afonso.script.sol/a-step-by-step-guide-to-the-delegate-pattern-in-swift-91a28de1baf8 for a more modern reference